core/src/main/java/com/linecorp/armeria/client/redirect/RedirectConfigBuilder.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.redirect;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.linecorp.armeria.internal.client.RedirectingClientUtil.allowAllDomains;
+import static java.util.Objects.requireNonNull;
+
+import java.net.URI;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.BiPredicate;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.client.AbstractClientOptionsBuilder;
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * A builder for {@link RedirectConfig}.
+ *
+ * @see AbstractClientOptionsBuilder#followRedirects(RedirectConfig)
+ */
+@UnstableApi
+public final class RedirectConfigBuilder {
+
+    private int maxRedirects = 19; // Widely used default value. https://stackoverflow.com/a/36041063/1736581
+
+    @Nullable
+    private Set<SessionProtocol> allowedProtocols;
+    private boolean isAllowingAllDomains;
+    @Nullable
+    private Set<String> allowedDomains;
+    @Nullable
+    private BiPredicate<ClientRequestContext, String> predicate;
+
+    RedirectConfigBuilder() {}
+
+    /**
+     * Sets the maximum number of automatic redirection that the client executes.
+     */
+    public RedirectConfigBuilder maxRedirects(int maxRedirects) {
+        checkArgument(maxRedirects > 0, "maxRedirects: %s (expected: > 0)", maxRedirects);
+        this.maxRedirects = maxRedirects;
+        return this;
+    }
+
+    /**
+     * Sets the {@link SessionProtocol}s that are allowed for automatic redirection.
+     * Only {@link SessionProtocol#HTTP} and {@link SessionProtocol#HTTPS} can be set.
+     *
+     * <p>When the allowed {@link SessionProtocol}s are not set, {@link SessionProtocol#HTTPS} is set
+     * by default. If the {@link WebClient} is created <b>with</b> a base URI that has
+     * {@link SessionProtocol#HTTP}, {@link SessionProtocol#HTTP} is also set.
```

Comment:
```suggestion
     * {@link SessionProtocol#HTTP}, {@link SessionProtocol#HTTPS} is also set.
```

Reply:
It's actually `{@link SessionProtocol#HTTP}`. 😄